htlcswitch/switch_test.go

```diff
@@ -66,6 +67,15 @@ func TestSwitchForward(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	s.addCircuit(&PaymentCircuit{
```

Comment:
Why is necessary to _manually_ add the payment circuit? It seems that the link the HTLC is being forwarded to should be populating it once it handles the downstream message.

Reply:
Because we are interacting with mock links, instead of the real links which are responsible for adding to the circuit map.